app/controllers/mixins/breadcrumbs_mixin.rb

```diff
@@ -33,33 +34,51 @@ def data_for_breadcrumbs
           breadcrumbs.push(:title => @title)
         end
       else
+        x_node_text_set if action_name == 'tree_select' # get text of the active node
+
         # Append breadcrumb from title of the accordion (eg "Policies")
-        breadcrumbs.push(:title => accord_title, :key => "#{accord_name}_accord", :action => "accordion_select") if accord_title
+        breadcrumbs.push(:title => accord_title, :key => "root") if accord_title
+
+        # On special page (tagitems, politems, etc.)
+        if @tagitems || @politems || @ownershipitems || @retireitems
+          breadcrumbs.push(special_page_breadcrumb(@tagitems || @politems || @ownershipitems || @retireitems, true)) unless options[:hide_special_item]
+          breadcrumbs.push(:title => @right_cell_text)
+        else
+          # Ancestry parents breadcrumbs (only in services)
+          breadcrumbs.concat(ancestry_parents(options[:ancestry], options[:record_info], options[:record_title])) if options[:ancestry]
 
-        # Append breadcrumbs created from the tree (eg "All policies > Red Hat policies > Policy 1")
-        breadcrumbs_from_tree = build_breadcrumbs_from_tree
-        breadcrumbs.concat(breadcrumbs_from_tree)
+          # Try to create treenode from the x_node
+          entity, entity_id = TreeBuilder.extract_node_model_and_id(x_node)
```

Comment:
This method returns 3 arguments and you probably want to ignore the last one, like this:
```suggestion
          entity, entity_id, _ = TreeBuilder.extract_node_model_and_id(x_node)
```
Otherwise `entity_id` is an array, which goes through the `find` properly if it has only one element. But it fails painfully if not :disappointed:

Reply:
Can it returns more IDs or does it always return an array with one id?

Reply:
It can and it can cause side effects, you should not destructure arguments to less than they are unless you're expecting this behavior. The `find` method will work properly with the `entity_id` not being an array.